naga/src/front/wgsl/parse/mod.rs

```diff
@@ -1182,6 +1200,9 @@ impl Parser {
                     ExpectedToken::Token(Token::Separator(',')),
                 ));
             }
+            // Save a lexer to be able to backtrack comments if need be.
+            let mut lexer_comments = lexer.clone();
```

Comment:
We can then just call `accumulate_doc_comments` here instead.

Reply:
The reason I went this route was because not all items should support comments, so if we try to accumulate on everything, there'd be a `Vec` instantiated to store the result, then we realize we don't need it, so we drop it. I suspect this would impact performance, but I didn't measure it.

Reply:
Empty `Vec`s don't allocate so I don't think it should be a problem.

Reply:
`accumulate_doc_comments` leaves the lexer in a "bad" state though, where it consumes an item "too much" in order to know it's stopped. I could change this behaviour to `peek` before consuming, or return the latest consumed item (which is not a comment) for further parsing :thinking:

Reply:
`accumulate_doc_item_comments` doesn't set `self.input = rest` when it encounters a token it can't consume and so it won't advance the tokenizer.